components-rs/telemetry.rs

```diff
@@ -227,54 +231,97 @@ pub unsafe extern "C" fn ddog_sidecar_telemetry_add_integration_log_buffer(
     buffer.buffer.push(SidecarAction::Telemetry(action));
 }
 
+pub struct ShmCache {
+    pub last_updated: Instant,
+    pub integrations: std::collections::HashSet<String>,
+    pub composer_paths: std::collections::HashSet<PathBuf>,
+    pub reader: OneWayShmReader<NamedShmHandle, CString>,
+}
+
+pub type ShmCacheMap = HashMap<(String, String), ShmCache>;
+
+#[no_mangle]
+pub extern "C" fn ddog_sidecar_telemetry_cache_new() -> Box<ShmCacheMap> {
+    ShmCacheMap::default().into()
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ddog_sidecar_telemetry_cache_drop(_: Box<ShmCacheMap>) {}
+
+unsafe fn ddog_sidecar_telemetry_cache_get_or_update(
+    cache: &mut ShmCacheMap,
+    service: CharSlice,
+    env: CharSlice,
+) -> *const ShmCache {
```

Comment:
```suggestion
) -> &ShmCache {
```
is there a particular reason to return that as pointer?